airflow/ti_deps/deps/prev_dagrun_dep.py

```diff
@@ -44,25 +44,16 @@ def _get_dep_statuses(self, ti, session, dep_context):
             return
 
         # Don't depend on the previous task instance if we are the first task
-        dag = ti.task.dag
-        if dag.catchup:
-            if dag.previous_schedule(ti.execution_date) is None:
-                yield self._passing_status(reason="This task does not have a schedule or is @once")
-                return
-            if dag.previous_schedule(ti.execution_date) < ti.task.start_date:
-                yield self._passing_status(
-                    reason="This task instance was the first task instance for its task."
-                )
-                return
+        dr = ti.get_dagrun(session=session)
+        if not dr:
+            last_dagrun = None
+        elif ti.task.dag.catchup:
+            last_dagrun = dr.get_previous_scheduled_dagrun(session)
         else:
-            dr = ti.get_dagrun(session=session)
-            last_dagrun = dr.get_previous_dagrun(session=session) if dr else None
-
-            if not last_dagrun:
-                yield self._passing_status(
-                    reason="This task instance was the first task instance for its task."
-                )
-                return
+            last_dagrun = dr.get_previous_dagrun(session=session)
+        if not last_dagrun:
+            yield self._passing_status(reason="This task instance was the first task instance for its task.")
+            return
 
         previous_ti = ti.get_previous_ti(session=session)
```

Comment:
This should be changed to get tis from `last_dagrun`, otherwise we are duplicating the effort to lookup the previous dag run twice.

Reply:
```suggestion
        previous_ti = last_dagrun.get_task_instance(ti.task_id, session=session)
```